src/FondOfSpryker/Yves/GoogleTagManageSessionConnector/Plugin/Variables/GoogleTagManagerTransactionIdPlugin.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+namespace FondOfSpryker\Yves\GoogleTagManagerSessionConnector\Plugin\Variables;
+
+use FondOfSpryker\Yves\GoogleTagManagerExtension\Dependency\GoogleTagManagerVariableBuilderPluginInterface;
+use Spryker\Yves\Kernel\AbstractPlugin;
+
+/**
+ * @method \FondOfSpryker\Yves\GoogleTagManagerSessionConnector\GoogleTagManagerSessionConnectorFactory getFactory()
+ */
+class GoogleTagManagerTransactionIdPlugin extends AbstractPlugin implements GoogleTagManagerVariableBuilderPluginInterface
+{
+    /**
+     * @param string $page
+     * @param array $params
+     *
+     * @return array
+     */
+    public function addVariable(string $page, array $params): array
+    {
+        return $this->getFactory()
+            ->createGoogleTagManagerSessionConnectorModel()
+            ->getTransactionId($page, $params);
```

Comment:
since array return... would it be better to name it getTransactionIds?

Reply:
it just returns an array as part of the google tag manager, not as response for multiple IDs

Reply:
Perhaps a transfer object for $page and $params could be better...